freespace2/freespace.cpp

```diff
@@ -2771,11 +2652,11 @@ void do_timing_test(float frame_time)
 		test_time = 0.0f;
 		Test_begin = 0;
 
-		for ( i = 0; i < NUM_MIXED_SOUNDS; i++ )
-			snds[i] = sound_handle::invalid();
+		for (auto & snd : snds)
+			snd = sound_handle::invalid();
 
 		// start looping digital sounds
-		for ( i = 0; i < NUM_MIXED_SOUNDS; i++ )
+		for (int i = 0; i < NUM_MIXED_SOUNDS; i++ )
```

Comment:
the last loop uses the index to access gamesnd_id(i) so it can't be a range based loop.

It should be possible to simply remove the ```snd= sound_handle::invalid();``` loop as the last loop overwrites all values anyways.